### ISSUE TITLE
Shared env file user with read only permissions cannot update current values

### ISSUE THREAD

**Issue by hiqqs** [a85/POSTMan-Chrome-Extension#10856]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

When you have an environment file set to read only permissions a user cannot simply update the initial values they have from changes that were made because it requires you to "reset all" and then save the env file which the user cannot do.

This seems like a gap, a user that is read only should still be able to "pull" the latest env file changes if you and reset or better yet be in sync with everyone else in the shared workspace

### Steps To Reproduce

1. Create an env file in a shared workspace
2. Set a user to read only on the env file
3. Make a change to the env file with an admin or user that can
4. Now with the read only user on the env try to update or "pull" the new env file changes that were made by the admin member

### Operating System

macOS

### Postman Version

9.15.13

### Postman Platform

Postman App

**Comment by DannyDainton** [comment]
Hey @johnhiggs

Could you provide a short recording of what you're experiencing, as you're describing something happening in the UI, it makes it easier for us to understand the issue.

Could you also clarify what value is being changed by the non read only user? The `Current Value` is not sync'd so that wouldn't be seen by anyone other than the person changing it in thier instance.

**Comment by hiqqs** [comment]
so the problem is that the read only user cannot save their version of the current value or "pull" the current value from the initials. does that make sense? @DannyDainton

**Comment by hiqqs** [comment]
as an admin, i can use the "reset all" feature on a shared env file which then i can save.  they cannot save their version of the env file to update their current values (or "pull" the latest initials)

**Comment by DannyDainton** [comment]
Just to be clear, you're not talking about a forked environment? You keep mentioning 'pull' the changes, which means something else in different contexts :D 

When the read-only user saves a current value, is an error notification shown? 

If the Save button is enabled and you go through the saving process, does the current value change to the newly added value?

A recording of this issue or screenshot will really help here, just trying to use some guesswork to try and know what you can see in front of you :D

**Comment by hiqqs** [comment]
im putting in quotes because as if you are pulling the latest updates to the environment variable for a read only user.  not an actual forked environment file.  

i can have a team member take a video if this doesn't make sense.

TLDR:

A shared environment file
An admin with read/write permission makes a change to a variable
A read only user for the environment file can not update their "current" values to see the latest change because they cannot save after using the reset all feature (user cannot write/save so when they run the shared collection it's not using the latest variable value for what's cached as their current)
So the outcome is they have to update their current values because they cannot "pull" or "reset" their current values and save the shared environment file

It's kind of a weird area i guess and may need more testing on our end

**Comment by hiqqs** [comment]
it's all sensitive information so i'd have to blur it out, perhaps you could send me a link to a shared workspace and we can replicate and test together

**Comment by coditva** [comment]
@johnhiggs Do you mean you're not able to update the click on the "save" button after resetting the environment?
Here are the steps I took:
1. On an environment where I only had `view` access, I tried to reset the current values to the initial values.
2. After that, I could see that the "Save" button was enabled which allows me to save the current values.
3. Clicking the save button saved the current values (although I did see a toast which says cannot save).


https://user-images.githubusercontent.com/8445768/165273719-7ee13605-ae2e-4841-b1cd-ed0c772b638d.mov
